axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/InvoicePaymentToolServiceSupplychainImpl.java

```diff
@@ -53,5 +53,9 @@ public void updateAmountPaid(Invoice invoice) throws AxelorException {
       // compute sale order totals
       Beans.get(SaleOrderComputeService.class)._computeSaleOrder(saleOrder);
     }
+    if (invoice.getPartner().getHasBlockedAccount()
+        && !invoice.getPartner().getHasManuallyBlockedAccount()) {
+      Beans.get(PartnerSupplychainService.class).updateBlockedAccount(invoice.getPartner());
```

Comment:
Please change injection in this class: change all injection from `Beans.get()` to injection by constructor